File: java_sdk/src/main/java/business_api_client/OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
/**
 * OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec
 */

@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2024-11-01T15:21:34.532937-07:00[America/Los_Angeles]")
public class OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec {
  @JsonProperty("audience_size")
  private String audienceSize = null;

  @JsonProperty("include_source")
  private Boolean includeSource = null;

  @JsonProperty("location_ids")
  private List<String> locationIds = null;

  @JsonProperty("mobile_os")
  private String mobileOs = null;

  @JsonProperty("placements")
  private List<String> placements = null;

  @JsonProperty("source_audience_id")
  private String sourceAudienceId = null;

  public OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec audienceSize(String audienceSize) {
    this.audienceSize = audienceSize;
    return this;
  }

   /**
   * Size of lookalike audience. Enum values: NARROW, BALANCED and BROAD.
   * @return audienceSize
  **/
  @Schema(description = "Size of lookalike audience. Enum values: NARROW, BALANCED and BROAD.")
  public String getAudienceSize() {
    return audienceSize;
  }

  public void setAudienceSize(String audienceSize) {
    this.audienceSize = audienceSize;
  }

  public OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec includeSource(Boolean includeSource) {
    this.includeSource = includeSource;
    return this;
  }

   /**
   * Whether to include the source audience in the new lookalike audience.
   * @return includeSource
  **/
  @Schema(description = "Whether to include the source audience in the new lookalike audience.")
  public Boolean isIncludeSource() {
    return includeSource;
  }

  public void setIncludeSource(Boolean includeSource) {
    this.includeSource = includeSource;
  }

  public OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec locationIds(List<String> locationIds) {
    this.locationIds = locationIds;
    return this;
  }

  public OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec addLocationIdsItem(String locationIdsItem) {
    if (this.locationIds == null) {
      this.locationIds = new ArrayList<String>();
    }
    this.locationIds.add(locationIdsItem);
    return this;
  }

   /**
   * IDs of the locations that you want to target. For enum values, see Enumeration-Lookalike Locations.
   * @return locationIds
  **/
  @Schema(description = "IDs of the locations that you want to target. For enum values, see Enumeration-Lookalike Locations.")
  public List<String> getLocationIds() {
    return locationIds;
  }

  public void setLocationIds(List<String> locationIds) {
    this.locationIds = locationIds;
  }

  public OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec mobileOs(String mobileOs) {
    this.mobileOs = mobileOs;
    return this;
  }

   /**
   * Device operating systems. Enum values support ALL for both ANDROID and IOS. See Enumeration-Audience Management-Lookalike System for details.
   * @return mobileOs
  **/
  @Schema(description = "Device operating systems. Enum values support ALL for both ANDROID and IOS. See Enumeration-Audience Management-Lookalike System for details.")
  public String getMobileOs() {
    return mobileOs;
  }

  public void setMobileOs(String mobileOs) {
    this.mobileOs = mobileOs;
  }

  public OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec placements(List<String> placements) {
    this.placements = placements;
    return this;
  }

  public OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec addPlacementsItem(String placementsItem) {
    if (this.placements == null) {
      this.placements = new ArrayList<String>();
    }
    this.placements.add(placementsItem);
    return this;
  }

   /**
   * The apps where you want to deliver your ads. Enum values: TikTok, TopBuzz &amp; BuzzVideo, Pangle. See Enumeration - Audience Management-Lookalike Placement for details.
   * @return placements
  **/
  @Schema(description = "The apps where you want to deliver your ads. Enum values: TikTok, TopBuzz & BuzzVideo, Pangle. See Enumeration - Audience Management-Lookalike Placement for details.")
  public List<String> getPlacements() {
    return placements;
  }

  public void setPlacements(List<String> placements) {
    this.placements = placements;
  }

  public OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec sourceAudienceId(String sourceAudienceId) {
    this.sourceAudienceId = sourceAudienceId;
    return this;
  }

   /**
   * Source (seed) audience ID. Source audience cannot be an existing lookalike audience. Note: The size of source audience should be no less than 100.
   * @return sourceAudienceId
  **/
  @Schema(description = "Source (seed) audience ID. Source audience cannot be an existing lookalike audience. Note: The size of source audience should be no less than 100.")
  public String getSourceAudienceId() {
    return sourceAudienceId;
  }

  public void setSourceAudienceId(String sourceAudienceId) {
    this.sourceAudienceId = sourceAudienceId;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec openApiv13dmpcustomAudiencelookalikecreateLookalikeSpec = (OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec) o;
    return Objects.equals(this.audienceSize, openApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.audienceSize) &&
        Objects.equals(this.includeSource, openApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.includeSource) &&
        Objects.equals(this.locationIds, openApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.locationIds) &&
        Objects.equals(this.mobileOs, openApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.mobileOs) &&
        Objects.equals(this.placements, openApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.placements) &&
        Objects.equals(this.sourceAudienceId, openApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.sourceAudienceId);
  }

  @Override
  public int hashCode() {
    return Objects.hash(audienceSize, includeSource, locationIds, mobileOs, placements, sourceAudienceId);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec {\n");
    
    sb.append("    audienceSize: ").append(toIndentedString(audienceSize)).append("\n");
    sb.append("    includeSource: ").append(toIndentedString(includeSource)).append("\n");
    sb.append("    locationIds: ").append(toIndentedString(locationIds)).append("\n");
    sb.append("    mobileOs: ").append(toIndentedString(mobileOs)).append("\n");
    sb.append("    placements: ").append(toIndentedString(placements)).append("\n");
    sb.append("    sourceAudienceId: ").append(toIndentedString(sourceAudienceId)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: js_sdk/src/model/OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient.js';

/**
 * The OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec model module.
 * @module model/OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec
 * @version 0.1.2
 */
export class OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec {
  /**
   * Constructs a new <code>OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec</code>.
   * @alias module:model/OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec
   * @class
   */
  constructor() {
  }

  /**
   * Constructs a <code>OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec} obj Optional instance to populate.
   * @return {module:model/OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec} The populated <code>OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec();
      if (data.hasOwnProperty('audience_size'))
        obj.audience_size = ApiClient.convertToType(data['audience_size'], 'String');
      if (data.hasOwnProperty('include_source'))
        obj.include_source = ApiClient.convertToType(data['include_source'], 'Boolean');
      if (data.hasOwnProperty('location_ids'))
        obj.location_ids = ApiClient.convertToType(data['location_ids'], ['String']);
      if (data.hasOwnProperty('mobile_os'))
        obj.mobile_os = ApiClient.convertToType(data['mobile_os'], 'String');
      if (data.hasOwnProperty('placements'))
        obj.placements = ApiClient.convertToType(data['placements'], ['String']);
      if (data.hasOwnProperty('source_audience_id'))
        obj.source_audience_id = ApiClient.convertToType(data['source_audience_id'], 'String');
    }
    return obj;
  }
}

/**
 * Size of lookalike audience. Enum values: NARROW, BALANCED and BROAD.
 * @member {String} audience_size
 */
OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.prototype.audience_size = undefined;

/**
 * Whether to include the source audience in the new lookalike audience.
 * @member {Boolean} include_source
 */
OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.prototype.include_source = undefined;

/**
 * IDs of the locations that you want to target. For enum values, see Enumeration-Lookalike Locations.
 * @member {Array.<String>} location_ids
 */
OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.prototype.location_ids = undefined;

/**
 * Device operating systems. Enum values support ALL for both ANDROID and IOS. See Enumeration-Audience Management-Lookalike System for details.
 * @member {String} mobile_os
 */
OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.prototype.mobile_os = undefined;

/**
 * The apps where you want to deliver your ads. Enum values: TikTok, TopBuzz & BuzzVideo, Pangle. See Enumeration - Audience Management-Lookalike Placement for details.
 * @member {Array.<String>} placements
 */
OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.prototype.placements = undefined;

/**
 * Source (seed) audience ID. Source audience cannot be an existing lookalike audience. Note: The size of source audience should be no less than 100.
 * @member {String} source_audience_id
 */
OpenApiv13dmpcustomAudiencelookalikecreateLookalikeSpec.prototype.source_audience_id = undefined;

